### PR TITLE
Update 1.26

### LINKED ISSUE #295: Add support for Kubernetes 1.26
**What would you like to be added**:
Since [OCI CCM released v1.26](https://github.com/oracle/oci-cloud-controller-manager/releases/tag/v1.26.0) we will need to update our configs. See https://github.com/oracle/cluster-api-provider-oci/pull/220 as example

**Why is this needed**:

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
Update CCM version to support 1.26 for tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #295

### CODE REVIEW

**Comment by joekr** [comment]
# unit tests

# e2e

```
Ran 7 of 27 Specs in 3080.411 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 20 Skipped
```

File: templates/cluster-template-oci-addons.yaml [hunk 1]
```diff
@@ -234,7 +234,7 @@ data:
                 path: /etc/kubernetes
           containers:
```
Comment (shyamradhakrishnan) [inline templates/cluster-template-oci-addons.yaml:235]: A snapshot-controller image has been added in [oci-csi-controller-driver.yaml](https://github.com/oracle/oci-cloud-controller-manager/releases/download/v1.25.0/oci-csi-controller-driver.yaml), can we add that please?

File: templates/cluster-template-oci-addons.yaml [hunk 2]
```diff
@@ -522,7 +522,7 @@ data:
                 - --fss-csi-endpoint=unix://var/run/shared-tmpfs/csi-fss.sock
               command:
                 - /usr/local/bin/oci-csi-controller-driver
-              image: ghcr.io/oracle/cloud-provider-oci:v1.25.0
+              image: ghcr.io/oracle/cloud-provider-oci:v1.26.0
```
Comment (shyamradhakrishnan) [inline templates/cluster-template-oci-addons.yaml:525]: the cloud-provider-oci image has to be replaced in 3 places right? I see only 2

File: test/e2e/data/ccm/ccm.yaml [hunk 3]
```diff
@@ -655,6 +661,21 @@ rules:
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
     verbs: ["patch"]
+  - apiGroups: [ "snapshot.storage.k8s.io" ]
+    resources: [ "volumesnapshotclasses" ]
```
Comment (shyamradhakrishnan) [inline test/e2e/data/ccm/ccm.yaml:665]: this change should go in the cluster-template-oci-addons file as well, and ccm.yaml

File: templates/cluster-template-oci-addons.yaml (hunk 1 above)
Comment (joekr) [reply templates/cluster-template-oci-addons.yaml:235]: The v1.26 link is https://github.com/oracle/oci-cloud-controller-manager/releases/download/v1.26.0/oci-csi-controller-driver.yaml that I'll be using.

File: test/e2e/data/ccm/ccm.yaml (hunk 3 above)
Comment (joekr) [reply test/e2e/data/ccm/ccm.yaml:665]: updated

File: templates/cluster-template-oci-addons.yaml [hunk 4]
```diff
@@ -848,7 +880,7 @@ data:
     roleRef:
       kind: ClusterRole
       name: csi-oci
-      apiGroup: rbac.authorization.k8s.io
+      apiGroup: rbac.authorization.k8s.io
```
Comment (shyamradhakrishnan) [inline templates/cluster-template-oci-addons.yaml:883]: extra space?

Comment (joekr) [reply templates/cluster-template-oci-addons.yaml]: fixed

**Review by shyamradhakrishnan — APPROVED** [review]

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**